LocalsTradeBot/Bots/EchoBot.cs

```diff
@@ -43,25 +56,35 @@ protected override async Task OnMessageActivityAsync(ITurnContext<IMessageActivi
             await _conversationState.SaveChangesAsync(turnContext, false, cancellationToken);
         }
 
-        private static async Task FillOutUserProfileAsync(Conversation conversation, ITurnContext turnContext, CancellationToken cancellationToken)
+        private async Task FillOutUserProfileAsync(Conversation conversation, ITurnContext turnContext, CancellationToken cancellationToken)
         {
             var input = turnContext.Activity.Text?.Trim();
             string message;
 
             switch (conversation.State)
             {
                 case Conversation.CurrentConversationState.Question:
-                    var responseMessage = "Thank you for your question. The question will be answered by administrator as soon as possible. " +
-                        "Please enter an email address to whom we will send a response";
-                    await turnContext.SendActivityAsync(responseMessage, null, null, cancellationToken);
-                    conversation.State = Conversation.CurrentConversationState.Email;
-                    break;
+                    if(ValidateQuestion(input, out var question, out message))
+                    {
+                        var responseMessage = "Thank you for your question. The question will be answered by administrator as soon as possible. " +
+                       "Please enter an email address to whom we will send a response";
+                        await turnContext.SendActivityAsync(responseMessage, null, null, cancellationToken);
+                        conversation.State = Conversation.CurrentConversationState.Email;
+                        conversation.Question = question;
```

Comment:
@MartynaUb Should not be question set to null, since it is already sent?
I mean, why are you keeping question at all, the bot will not answer it himself.

Reply:
Question will be sent in the later stage where email is provided, at the moment it is stored in memory until email is provided and only then I send it to your service and reset email value to null